### PR TITLE
Add fastpca

### PR DESCRIPTION
**Pull request by simonbray**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/fastpca/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+build:
+  number: 0
+  skip: True  # [win or osx]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - boost
```
Comment (bgruening) [inline recipes/fastpca/meta.yaml]: this should probably be a host dependency and maybe just boost-cpp

File: recipes/fastpca/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  host:
+    - libopenblas
```
Comment (bgruening) [inline recipes/fastpca/meta.yaml]: can you follow https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=blas#blas

File: recipes/fastpca/meta.yaml (hunk 1 above)
Comment (simonbray) [reply recipes/fastpca/meta.yaml]: I tried it in host first and the build fails.

Comment (simonbray) [reply recipes/fastpca/meta.yaml]: can try boost-cpp

File: recipes/fastpca/meta.yaml (hunk 2 above)
Comment (simonbray) [reply recipes/fastpca/meta.yaml]: yes, thanks

**Comment by simonbray** [comment]
With boost-cpp or boost under host I get:
```
[ 90%] Building CXX object src/CMakeFiles/fastpca.dir/fastpca.cpp.o
/home/conda/staged-recipes/build_artifacts/fastpca_1574274965824/work/src/fastpca.cpp:35:10: fatal error: boost/program_options.hpp: No such file or directory
 #include <boost/program_options.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

File: recipes/fastpca/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+build:
+  number: 0
+  skip: True  # [win or osx]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - boost-cpp
```
Comment (scopatz) [inline recipes/fastpca/meta.yaml]: boost-cpp should be a host dep, not a build dep

File: recipes/fastpca/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  host:
+    - libblas
+    - liblapack
+
+  run:
+    - boost
```
Comment (scopatz) [inline recipes/fastpca/meta.yaml]: Also, this should be boost-cpp.  `boost` includes boost-python stupidly

File: recipes/fastpca/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - boost-cpp
+
+  host:
+    - libblas
+    - liblapack
```
Comment (scopatz) [inline recipes/fastpca/meta.yaml]: Please follow these instructions for blas: https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=mesa#blas

**Pull request merged**